### PR TITLE
Add S3 Bucket Should Have Bucket Policy query for CloudFormation #1073

### PR DESCRIPTION
**Pull request by pedro-mimoso**
Checks if S3 Bucket has the same name as a Bucket Policy, if it has, S3 Bucket has a Bucket Policy associated #1073

### CODE REVIEW

File: assets/queries/cloudFormation/s3_bucket_should_have_bucket_policy/test/negative.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Resources:
+  S3Bucket:
+    Type: 'AWS::S3::Bucket'
+    DeletionPolicy: Retain
+    Properties:
+      BucketName: DOC-EXAMPLE-BUCKET
+  SampleBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: DOC-EXAMPLE-BUCKET
```
Comment (joaoReigota1) [inline assets/queries/cloudFormation/s3_bucket_should_have_bucket_policy/test/negative.yaml:12]: Bucket name should be lowercase. Must fit this regex '^[a-z0-9][a-z0-9.-]*[a-z0-9]$'

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]
Has conflicts -> git checkout master -> git pull -> git checkout 
feature/add_query_s3_bucket_should_have_bucket_policy -> git pull -> git merge master -> fix conflicts

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]
942 files were changed something with the merge was wrong

**Comment by joaoReigota1** [comment]
Approved

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**